GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -136,24 +117,30 @@ public async Task UpdateSubmodulesStructureAsync(string workingDirectory, string
         /// <inheritdoc />
         public async Task UpdateSubmodulesStatusAsync(string workingDirectory, [CanBeNull] IReadOnlyList<GitItemStatus> gitStatus, bool forceUpdate)
         {
-            if (_submoduleInfoResult == null)
+            if (gitStatus == null)
             {
-                _gitStatusWhileUpdatingStructure = gitStatus;
                 return;
             }
 
-            // Throttle updates triggered from status updates
-            TimeSpan elapsed = DateTime.Now - _previousSubmoduleUpdateTime;
-            if (gitStatus == null || (!forceUpdate && elapsed.TotalSeconds <= 15))
+            var cancelToken = _submodulesStatusSequence.Next();
+            await TaskScheduler.Default;
+            cancelToken.ThrowIfCancellationRequested();
+
+            while (_submoduleInfoResult == null)
             {
-                Console.WriteLine($"{MethodBase.GetCurrentMethod().Name} called too early again - aborting");
-                return;
+                // Wait until the structure is built
+                await Task.Delay(TimeSpan.FromMilliseconds(300), cancelToken);
+                cancelToken.ThrowIfCancellationRequested();
             }
 
-            var cancelToken = _submodulesStatusSequence.Next();
-            _gitStatusWhileUpdatingStructure = null;
-
-            await TaskScheduler.Default;
+            // Throttle updates triggered from status updates
+            const int minRefreshInterval = 15;
```

Comment:
I guess, Igor meant to create a class constant, which is visible at the top of the file.